test/p3_tests.jl

```diff
@@ -609,8 +611,141 @@ function test_p3_bulk_liquid_ice_collisions(FT)
+        @test QCCOL == QRCOL
+        @test NCCOL == NRCOL
+        @test BCCOL == BRCOL
+
+        # Test edge cases for full collision integration
+
+        # Zero ice content
+        n_i_zero(Dᵢ) = FT(0)
+        rates = P3.∫liquid_ice_collisions(
+            n_i_zero, ∂ₜM_max, cloud_integrals, rain_integrals, ice_bounds,
+        )
+        @test all(iszero, rates)
+
+        # Zero liquid content
+        n_zero = Returns(FT(0))
+        zero_liq_integrals = P3.get_liquid_integrals(n_zero, ∂ₜV, m_l, ρ′_rim, liq_bounds)
+        rates = P3.∫liquid_ice_collisions(
+            n_i, ∂ₜM_max, zero_liq_integrals, zero_liq_integrals, ice_bounds,
+        )
+        @test all(iszero, rates)
+
+        # No freezing (above freezing temperature)
+        ∂ₜ𝓜_max_zero(Dᵢ) = FT(0)
+        rates = P3.∫liquid_ice_collisions(
+            n_i, ∂ₜ𝓜_max_zero, cloud_integrals, rain_integrals, ice_bounds,
+        )
+        QCCOL, NCCOL, QRCOL, NRCOL, ∫𝓜_col, BCCOL, BRCOL, ∫𝟙_wet_𝓜_col = rates
+        @test QCCOL == 0  # No cloud freezing
+        @test QRCOL == 0  # No rain freezing
+        @test ∫𝓜_col > 0  # But collisions still occur
+        @test ∫𝟙_wet_𝓜_col == ∫𝓜_col  # All collisions result in wet growth
+    end
 
+    @testset "Bulk liquid-ice collisions" begin
+        # Test the high-level interface with real P3 parameters
+        state = P3.get_state(params; F_rim, ρ_rim, L_ice = Lᵢ, N_ice = Nᵢ)
+        logλ = P3.get_distribution_logλ(state)
+
+        # Create mock particle size distributions
+        toml_dict = CP.create_toml_dict(FT)
+        psd_c = CMP.CloudParticlePDF_SB2006(toml_dict)
+        psd_r = CMP.RainParticlePDF_SB2006_limited(toml_dict)
+
+        # Test parameters
+        L_c = FT(1e-3)  # 1 g/m³ cloud water
+        N_c = FT(1e8)   # 100 million cloud droplets per m³
+        L_r = FT(1e-4)  # 0.1 g/m³ rain water
+        N_r = FT(1e6)   # 1 million raindrops per m³
+        T = T_freeze - FT(5)  # 5K below freezing
+
+        # Liquid particle mass function
+        ρw = psd_c.ρw
+        m_l(Dₗ) = ρw * CO.volume_sphere_D(Dₗ)
+
+        # Test the high-level interface
+        rates = P3.∫liquid_ice_collisions(
+            state, logλ, psd_c, psd_r, L_c, N_c, L_r, N_r,
+            aps, tps, vel_params, ρₐ, T, m_l,
+        )
+
+        QCCOL, NCCOL, QRCOL, NRCOL, ∫𝓜_col, BCCOL, BRCOL, ∫𝟙_wet_𝓜_col = rates
+
+        # Basic sanity checks
+        @test all(rates .>= 0)
+        @test QCCOL + QRCOL <= ∫𝓜_col
+        @test ∫𝟙_wet_𝓜_col <= ∫𝓜_col
+
+        # Test that rates scale with liquid content
+        rates_2x = P3.∫liquid_ice_collisions(
+            state, logλ, psd_c, psd_r, 2 * L_c, N_c, 2 * L_r, N_r,
+            aps, tps, vel_params, ρₐ, T, m_l,
+        )
+
+        # Rates should approximately double (not exact due to size distribution changes)
+        @test rates_2x[1] ≈ 2 * QCCOL rtol = 0.5  # Allow 50% tolerance for distribution effects
+        @test rates_2x[3] ≈ 2 * QRCOL rtol = 0.5
+    end
```

Comment:
I wonder if it would also be useful to add a "smoke test". I.e. save a couple of reference values for collision rates and just test if they are unchanged as we modify the code. I know it's not ideal, but it's good to have a warning if things change without us intending to change them